app/controllers/tests_controller.rb

```diff
@@ -0,0 +1,29 @@
+class TestsController < ApplicationController
+  def show
+    @test = Test.find(params[:id])
+    start_test @test
+  end
+
+  def answer
+    @question = Question.find(params[:question])
+    @is_right = (@question.right_answer == params[:ans].to_i)
+    if @is_right
+      @test_result = current_user.test_results.find_by(test_id: params[:test])
+      @test_result.result = @test_result.result + 1
+      @test_result.save
+    end
+    respond_to do |format|
+      format.js
```

Comment:
That's a bad and ugly way, better render JSON here and handle it on the client.

Nobody renders JS on the backend today, only 1 device supports it - browser. Mobile devices, desktop devices - none of them support it.